lib/SaferpayJson/Request/RequestConfig.php

```diff
@@ -5,21 +5,45 @@
 namespace Ticketpark\SaferpayJson\Request;
 
 use GuzzleHttp\Client;
+use InvalidArgumentException;
 
 final class RequestConfig
 {
+    public const MIN_RETRY_INDICATOR = 0;
+    public const MAX_RETRY_INDICATOR = 9;
+
     private string $apiKey;
     private string $apiSecret;
     private string $customerId;
     private bool $test;
     private ?Client $client = null;
+    private ?string $requestId;
+    private int $retryIndicator;
 
-    public function __construct(string $apiKey, string $apiSecret, string $customerId, bool $test = false)
-    {
+    public function __construct(
+        string  $apiKey,
+        string  $apiSecret,
+        string  $customerId,
+        bool    $test = false,
+        ?string $requestId = null,
+        int     $retryIndicator = 0
+    ) {
```

Comment:
I understand the reason, but I am not very happy with the `$test` parameter not being the last one anymore.

Would it be worth to remove optional parameters `requestId` an `retryIndicator` from the constructor and add a separate call instead?

Something like:

```php
$requestConfig = (new RequestConfig(
    $apiKey,
    $apiSecret,
    $customerId
))
->setRequestId('some-string')
->setRetryIndicator(0);
```

Reply:
Yes I was also struggling a bit as to whether the values ​​should really only be set via the constructor. But I think your suggestion is definitely an improvement, because it means the request config can be reused and only the variable part (RequestID, RetryIndicator) can be set at runtime.